src/core/reactor.cc

```diff
@@ -4444,7 +4444,16 @@ reactor::init_new_scheduling_group_key(scheduling_group_key key, scheduling_grou
 }
 
 future<>
-reactor::destroy_scheduling_group(scheduling_group sg) {
+reactor::destroy_scheduling_group(scheduling_group sg) noexcept {
+    if (sg._id >= max_scheduling_groups() || !_task_queues[sg._id]) {
+        auto msg = format("Invalid scheduling_group {}", sg._id);
+        on_internal_error_noexcept(seastar_logger, msg);
+#ifndef SEASTAR_DEBUG
+        abort();
+#else
+        return make_exception_future<>(std::invalid_argument(msg));
```

Comment:
Why the different behavior?

Reply:
Based on your reply on email to a similar patch for the smp service group:
```
On Sun, 2021-11-28 at 11:48 +0200, Avi Kivity wrote:

> This is similar to EBADF. If you can attempt to destroy an out-of-range 
> smp service group, you can attempt to destroy an in-use smp service 
> group that's unrelated to you. 
> Returning an exception is hiding a 
> critical error.
> 
> 
> Unrelated, destroy_smp_service_group is a close()-like function that is 
> used on shutdown paths. It's often not possible to recover from an error 
> there.
> 
> 
> I think on_internal_error_noexcept() is a better choice here.
```

However, this can't be tested if we abort uncoditionally, hence the different behavior in debug mode, to allow testing the code.

Reply:
It makes debug mode for users worse, since it delays error reporting. Debug mode should provide a better debug experience than release.